tests/detectors/test_queued_detector.py

```diff
@@ -116,14 +146,92 @@ async def test_detector_can_detect_twice(self):
                 detection_error=False,
             ),
         )
+        await self.detector_can_detect_twice(ctx)
 
-        detector = QueuedDetector(ctx, 1, [])
-        expect(detector).not_to_be_null()
 
-        data = await detector.detect()
-        expect(data).to_be_empty()
-        expect(ctx.request.detection_error).to_be_false()
-        expect(detector.queue).not_to_be_null()
+class QueuedSentinelDetectorTestCase(SharedQueuedDetectorTestCase):
```

Comment:
```suggestion
@pytest.fixture
def ctx_mock(request):
    request.cls.ctx = mock.Mock(
        config=mock.Mock(
            REDIS_QUEUE_MODE=TEST_REDIS_SENTINEL_MODE,
            REDIS_QUEUE_SENTINEL_INSTANCES=TEST_REDIS_QUEUE_SENTINEL_INSTANCES,
            REDIS_QUEUE_SENTINEL_MASTER_INSTANCE=TEST_REDIS_SENTINEL_MASTER_INSTANCE,
            REDIS_QUEUE_SENTINEL_MASTER_PASSWORD=TEST_REDIS_PASSWORD,
            REDIS_QUEUE_SENTINEL_MASTER_DB=0,
            REDIS_QUEUE_SENTINEL_SOCKET_TIMEOUT=TEST_REDIS_SENTINEL_SOCKET_TIMEOUT,
            REDIS_QUEUE_SENTINEL_PASSWORD=TEST_REDIS_PASSWORD,
        ),
        request=mock.Mock(
            image_url="/image/test.jpg",
            detection_error=False,
        ),
    )


@pytest.mark.usefixtures("ctx_mock")
class QueuedSentinelDetectorTestCase(SharedQueuedDetectorTestCase):
```

This is going to help us reduce repeated code 🎉 — which is undesirable, even in tests.

Once this PR is merged, one of us should create another PR replicating the same changes to the `QueuedDetectorTestCase` class.

Reply:
Looking at it again, it could maybe be part of `setup` instead of a fixture. I started with the fixture and never went back 😹 

WDYT, folks?

Reply:
So I changed it to `setUp` and removed duplicate codes.

Reply:
Great! 👍 That's better! I guess I got too used to fixtures ☺️